pkgs/development/compilers/llvm/14/clang/default.nix

```diff
@@ -26,6 +26,9 @@ let
 
     buildInputs = [ libxml2 libllvm ];
 
+    # workaround https://github.com/NixOS/nixpkgs/issues/201254
+    NIX_LDFLAGS = if stdenv.isLinux && stdenv.isAarch64 && stdenv.cc.isGNU then "-lgcc" else null;
```

Comment:
We should consider if adding an argument above like `{ withLibgcc ? stdenv.isLinux && stdenv.isAarch64 && stdenv.cc.isGNU, ... }:` is better than putting these conditions inline in the body of derivations. This could make overriding LLVM easier and I think its self-documentation to have code in the form of `effect ? cause` rather than `if ${cause} then ${action} else null`.

Reply:
Since this is just an ugly workaround, I would prefer to keep this as ugly as possible. :grimacing: